test/index.test.js

```diff
@@ -0,0 +1,443 @@
+'use strict'
+
+const test = require('tap').test
+const NoGres = require('../index')
+
+test('client', t => {
+  t.test('expect', t => {
```

Comment:
no need to nest just write a lot of top level tests.

Reply:
You can also use multiple files!

Reply:
I've removed 1 level of nesting for readability. Now only `client` and `pool` are nested. Can use separate files for sure, but maybe when it would grow a bit larger?